### PR TITLE
Add unit tests for calling `query()` with empty predicate view on search trees

### PR DESCRIPTION
**Pull request by dalg24**
```C++
search_tree.query( queries, indices, offset, ranks );
```
In #303, Bruno was checking whether `offset.extent( 0 )` is greater than zero which by design should always be true.
Say `queries` has size `n_queries` then:
* `offset` will be of size `n_queries + 1`
* `indices` and `ranks` have size `offset( n_queries )`

This is true even when`n_queries` is zero in which case `indices` and `ranks` are empty and `offset` contains a single `0` element.

Hence, it is always valid to write something like this to iterate through the results of the search for all queries:
```C++
for (int q = 0; q < n_queries; ++q )
  for (int i = offset( q ); i < offset( q + 1 ); ++i)
    // etc.
```

When reviewing the interpolation PR, I realized we were not actually testing for that, so here it is :)

### CODE REVIEW

File: packages/Search/test/tstDistributedSearchTree.cpp [hunk 1]
```diff
@@ -146,6 +146,86 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( DistributedSearchTree, hello_world,
...
+    auto indices_host = Kokkos::create_mirror_view( indices );
+    deep_copy( indices_host, indices );
+    auto offset_host = Kokkos::create_mirror_view( offset );
+    deep_copy( offset_host, offset );
+    auto ranks_host = Kokkos::create_mirror_view( ranks );
+    deep_copy( ranks_host, ranks );
+
+    TEST_COMPARE_ARRAYS( indices_host, indices_ref );
+    TEST_COMPARE_ARRAYS( offset_host, offset_ref );
+    TEST_COMPARE_ARRAYS( ranks_host, ranks_ref );
```
Comment (Rombur) [inline packages/Search/test/tstDistributedSearchTree.cpp]: No test for `extent`?

File: packages/Search/test/tstDistributedSearchTree.cpp [hunk 2]
```diff
@@ -146,6 +146,86 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( DistributedSearchTree, hello_world,
     }
 }
 
+template <typename Query, typename DeviceType>
+void check_results( DataTransferKit::DistributedSearchTree<DeviceType> &tree,
+                    Kokkos::View<Query *, DeviceType> const &queries,
+                    std::vector<int> const &indices_ref,
+                    std::vector<int> const &offset_ref,
+                    std::vector<int> const &ranks_ref, bool &success,
+                    Teuchos::FancyOStream &out )
```
Comment (Rombur) [inline packages/Search/test/tstDistributedSearchTree.cpp]: Where do you use `success` and `out`?

File: packages/Search/test/tstLinearBVH.cpp [hunk 3]
```diff
@@ -19,6 +19,26 @@
 #include <random>
 #include <tuple>
 
+template <typename Query, typename DeviceType>
+void check_results( DataTransferKit::BVH<DeviceType> &bvh,
+                    Kokkos::View<Query *, DeviceType> const &queries,
+                    std::vector<int> const &indices_ref,
+                    std::vector<int> const &offset_ref, bool &success,
+                    Teuchos::FancyOStream &out )
```
Comment (Rombur) [inline packages/Search/test/tstLinearBVH.cpp]: Where do you use `success` and `out`?

File: packages/Search/test/tstLinearBVH.cpp [hunk 4]
```diff
@@ -204,6 +204,26 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( LinearBVH, nearest_queries, DeviceType )
     TEST_EQUALITY( offset.extent( 0 ), 2 );
 }
 
+template <typename Query, typename DeviceType>
+void check_results( DataTransferKit::BVH<DeviceType> &bvh,
+                    Kokkos::View<Query *, DeviceType> queries,
+                    std::vector<int> const &indices_ref,
+                    std::vector<int> const &offset_ref, bool &success,
+                    Teuchos::FancyOStream &out )
```
[the remark this answers is not in the document]
Comment (dalg24) [reply packages/Search/test/tstLinearBVH.cpp]: They come from the Teuchos unit test macro expansion.  They are necessary for calling test assertion macros.

[the remark this answers is not in the document]
Comment (Rombur) [reply packages/Search/test/tstLinearBVH.cpp]: OK can you jut add a comment

File: packages/Search/test/tstLinearBVH.cpp [hunk 5]
```diff
@@ -204,6 +204,26 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( LinearBVH, nearest_queries, DeviceType )
...
+{
+    Kokkos::View<int *, DeviceType> indices( "indices" );
+    Kokkos::View<int *, DeviceType> offset( "offset" );
+    bvh.query( queries, indices, offset );
+
+    auto indices_host = Kokkos::create_mirror_view( indices );
+    deep_copy( indices_host, indices );
+    auto offset_host = Kokkos::create_mirror_view( offset );
+    deep_copy( offset_host, offset );
+
```
[the remark this answers is not in the document]
Comment (dalg24) [reply packages/Search/test/tstLinearBVH.cpp]: View are one-dimensional so `TEST_COMPARE_ARRAYS` which compares output of `size()` before comparing element by element takes care of it.

[the remark this answers is not in the document]
Comment (Rombur) [reply packages/Search/test/tstLinearBVH.cpp]: ok

File: packages/Search/test/tstDistributedSearchTree.cpp (hunk 2 above)
Comment (dalg24) [reply packages/Search/test/tstDistributedSearchTree.cpp]: I added a comment.
